tests/foreman/destructive/test_katello_agent.py

```diff
@@ -167,14 +166,11 @@ def test_positive_upgrade_warning(sat_with_katello_agent):
         'which will automatically uninstall katello-agent.'
     )
 
-    upstream_rpms = sat.get_repo_files_by_url(constants.FOREMAN_NIGHTLY_URL)
-    fm_rpm = [rpm for rpm in upstream_rpms if 'foreman_maintain' in rpm]
-    assert fm_rpm, 'No upstream foreman-maintain package found'
-
-    for rpm in fm_rpm:
-        res = sat.execute(f'yum -y install {constants.FOREMAN_NIGHTLY_URL}{rpm}')
-        assert res.status == 0, f'{rpm} installation failed'
+    settings.repos.satellite_version_undr = '6_15'
+    settings.repos.snap_version_undr = '6_15_0-13_0'
+    sat.create_custom_repos(next_maintain=settings.repos.NEW_MAINTENANCE_REPO)
```

Comment:
I don't think this is the right way to test this scenario. We're using the same foreman-maintain version over and over here. Can we at least get the latest foreman-maintain package from the latest 6.15.z?

Reply:
@jameerpathan111 Hmm, true, that would be much better. 
Is there some nice way to check the latest `6.15.z` while running `6.14.z`? The only way I can see now is just check the dogfood server and find the latest one published there.

Reply:
```
ver = sat.version.split('.')
target_ver = f'{ver[0]}.{int(ver[1]) + 1}'
```
```
In [4]: maintain_repo = ohsnap.dogfood_repository(
   ...:         settings.ohsnap,
   ...:         product='satellite',
   ...:         repo='maintenance',
   ...:         release=target_ver,
   ...:         os_release=settings.server.version.rhel_version,
   ...:     )
```
This should give the latest 6.15.z maintenance repo url which you can later use as `sat.create_custom_repos(next_maintain=maintain_repo.baseurl)`

Reply:
Maybe @ogajduse knows a better way to do this:)

Reply:
This one seems to work well.

Reply:
great:)